x-pack/test/security_solution_ftr/page_objects/detections/index.ts

```diff
@@ -0,0 +1,157 @@
+  const testSubjects = getService('testSubjects');
+
+  class DetectionsPage {
+    async navigateHome(): Promise<void> {
+      await this.navigateToDetectionsPage();
+    }
+
+    async navigateToRules(): Promise<void> {
+      await this.navigateToDetectionsPage('rules');
+    }
+
+    async navigateToRuleMonitoring(): Promise<void> {
+      await common.clickAndValidate('allRulesTableTab-monitoring', 'monitoring-table');
+    }
+
+    async navigateToExceptionList(): Promise<void> {
+      await common.clickAndValidate('allRulesTableTab-exceptions', 'exceptions-table');
+    }
+
+    async navigateToCreateRule(): Promise<void> {
+      await this.navigateToDetectionsPage('rules/create');
+    }
+
+    async addEqlQuery(query: string): Promise<void> {
+      const textArea = await find.byCssSelector(
+        '[data-test-subj="detectionEngineStepDefineRuleEqlQueryBar"] [data-test-subj="eqlQueryBarTextInput"]'
+      );
+      await textArea.type(query);
+    }
+
+    async replaceIndexPattern(): Promise<void> {
+      const buttons = await find.allByCssSelector('[data-test-subj="comboBoxInput"] button');
+      await buttons.map(async (button: WebElementWrapper) => await button.click());
+      const input = await find.byCssSelector('input[data-test-subj="comboBoxSearchInput"]');
+      await input.type('*');
+    }
+
+    async openImportQueryModal(): Promise<void> {
+      await testSubjects.click('importQueryFromSavedTimeline');
+      await testSubjects.exists('open-timeline-modal-body-filter-default');
+    }
+
+    async viewTemplatesInImportQueryModal(): Promise<void> {
+      await common.clickAndValidate('open-timeline-modal-body-filter-template', 'timelines-table');
+    }
+
+    async closeImportQueryModal(): Promise<void> {
+      await find.clickByCssSelector('.euiButtonIcon.euiButtonIcon--text.euiModal__closeIcon');
+    }
+
+    async selectMachineLearningJob(): Promise<void> {
+      const jobDropdown = await find.byCssSelector('[data-test-subj="mlJobSelect"] button');
+      await jobDropdown.click();
+      const firstJob = await find.byCssSelector('#high_distinct_count_error_message');
+      await firstJob.click();
+    }
+
+    async openAddFilterPopover(): Promise<void> {
+      const addButtons = await find.allByCssSelector('[data-test-subj="addFilter"]');
+      await addButtons[1].click();
+      await testSubjects.exists('saveFilter');
+    }
+
+    async closeAddFilterPopover(): Promise<void> {
+      const closeButton = await testSubjects.find('cancelSaveFilter');
+      await closeButton.click();
+    }
+
+    async toggleFilterActions(): Promise<void> {
+      const filterActions = await find.allByCssSelector('[data-test-subj="addFilter"]');
+      await filterActions[1].click();
+    }
+
+    async toggleSavedQueries(): Promise<void> {
+      const filterActions = await find.allByCssSelector(
+        '[data-test-subj="saved-query-management-popover-button"]'
+      );
+      await filterActions[1].click();
```

Comment:
```suggestion
      const filterActions = await testSubjects.findAll('saved-query-management-popover-button"');
      await filterActions[1].click();
```

Reply:
Why do we click specifically the 2nd element?

Reply:
because there seems to be three of these buttons, and clicking the second one opens the popover

Reply:
Got it. Could we set unique data-test-subj based on the action?